source/validate_ext_inst.cpp

```diff
@@ -0,0 +1,755 @@
+      case GLSLstd450InverseSqrt:
+      case GLSLstd450FMin:
+      case GLSLstd450FMax:
+      case GLSLstd450FClamp:
+      case GLSLstd450FMix:
+      case GLSLstd450Step:
+      case GLSLstd450SmoothStep:
+      case GLSLstd450Fma:
+      case GLSLstd450Normalize:
+      case GLSLstd450FaceForward:
+      case GLSLstd450Reflect:
+      case GLSLstd450NMin:
+      case GLSLstd450NMax:
+      case GLSLstd450NClamp: {
+        if (!_.IsFloatScalarOrVectorType(result_type)) {
+          return _.diag(SPV_ERROR_INVALID_DATA)
+                 << ext_inst_name() << ": "
+                 << "expected Result Type to be a float scalar or vector type";
+        }
+
+        for (uint32_t operand_index = 4; operand_index < num_operands;
+             ++operand_index) {
+          const uint32_t operand_type = _.GetOperandTypeId(inst, operand_index);
+          if (result_type != operand_type) {
+            return _.diag(SPV_ERROR_INVALID_DATA)
+                   << ext_inst_name() << ": "
+                   << "expected types of all operands to be equal to Result "
+                      "Type";
+          }
+        }
+        break;
+      }
+
+      case GLSLstd450SAbs:
+      case GLSLstd450SSign:
+      case GLSLstd450UMin:
+      case GLSLstd450SMin:
+      case GLSLstd450UMax:
+      case GLSLstd450SMax:
+      case GLSLstd450UClamp:
+      case GLSLstd450SClamp:
+      case GLSLstd450FindILsb:
+      case GLSLstd450FindUMsb:
+      case GLSLstd450FindSMsb: {
+        if (!_.IsIntScalarOrVectorType(result_type)) {
+          return _.diag(SPV_ERROR_INVALID_DATA)
+                 << ext_inst_name() << ": "
+                 << "expected Result Type to be an int scalar or vector type";
+        }
+
+        const uint32_t result_type_bit_width = _.GetBitWidth(result_type);
+        const uint32_t result_type_dimension = _.GetDimension(result_type);
+
+        for (uint32_t operand_index = 4; operand_index < num_operands;
+             ++operand_index) {
+          const uint32_t operand_type = _.GetOperandTypeId(inst, operand_index);
+          if (!_.IsIntScalarOrVectorType(operand_type)) {
+            return _.diag(SPV_ERROR_INVALID_DATA)
+                   << ext_inst_name() << ": "
+                   << "expected all operands to be int scalars or vectors";
+          }
+
+          if (result_type_dimension != _.GetDimension(operand_type)) {
+            return _.diag(SPV_ERROR_INVALID_DATA)
+                   << ext_inst_name() << ": "
+                   << "expected all operands to have the same dimension as "
+                   << "Result Type";
+          }
+
+          if (result_type_bit_width != _.GetBitWidth(operand_type)) {
+            return _.diag(SPV_ERROR_INVALID_DATA)
+                   << ext_inst_name() << ": "
+                   << "expected all operands to have the same bit width as "
+                   << "Result Type";
+          }
+
+          if (ext_inst_key == GLSLstd450FindILsb ||
```

Comment:
FindILsb does not have this restriction.
(Was confirmed by Khronos internal issue #173 if you want to look it up.)

Reply:
It seems that the issue was left undecided.

Reply:
We decided to leave ILsb as it is currently specified, without the constraint.
The undecided part was whether to relax the UMsb SMsb instructions.   We're waiting for a request change it.